### PR TITLE
introduce sql query support

### PR DESCRIPTION
**Pull request by adecaro**

### CODE REVIEW

File: token/services/db/sql/common/query.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+const (
+	SelectStatement         = `SELECT`
+	SelectDistinctStatement = `SELECT DISTINCT`
+	InsertStatement         = `INSERT INTO`
+	UpdateStatement         = `UPDATE`
+)
+
+type Select struct {
+	stmt    string
+	columns []string
```
Comment (alexandrosfilios) [inline token/services/db/sql/common/query.go:23]: I would add optionally the table name for each column

File: token/services/db/sql/common/query.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+	from    []string
+	where   string
```
Comment (alexandrosfilios) [inline token/services/db/sql/common/query.go:25]: Where can be a *common.Condition

File: token/services/db/sql/common/query.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+	orderBy string
```
Comment (alexandrosfilios) [inline token/services/db/sql/common/query.go:26]: order by could be a slice of Column

File: token/services/db/sql/common/query.go [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+	s.from = tables
+	return s
+}
+
+func (s *Select) Where(where string) *Select {
+	s.where = where
+	return s
+}
+
+func (s *Select) Compile() (string, error) {
```
Comment (alexandrosfilios) [inline token/services/db/sql/common/query.go:53]: I would leave compile to an interpreter. Different SQL implementations have different results (e.g. check the common Interpreter and Postgres Interpreter). So each db implementation could be injected its own interpreter like is already done for the conditions.
```
type Interpreter interface {
 Select(s Select) (string, error)
 Insert(I Insert) (string, error)
}
```

File: token/services/db/sql/common/query.go [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+}
+
+func NewInsertInto(table string) *Insert {
+	return &Insert{
+		stmt:  InsertStatement,
+		table: table,
+	}
+}
+
+func (i *Insert) Rows(rows string) *Insert {
```
Comment (alexandrosfilios) [inline token/services/db/sql/common/query.go:99]: This refers to fields/columns I think. Also we could pass a slice to make it clear.

**Review by alexandrosfilios — COMMENTED** [review]
Some suggestions

**Comment by adecaro** [comment]
@alexandrosfilios , thanks for the comment. Let's address them in another PR when time will come :)

**Review by alexandrosfilios — APPROVED** [review]
We can address these comments at a later stage. LGTM

**Pull request merged**